JASP-R-Interface/jaspResults/R/zzzWrappers.R

```diff
@@ -192,7 +192,8 @@ jaspResultsR <- R6Class(
     getOtherObjectsForState = function()        private$jaspObject$getOtherObjectsForState()
 	),
 	active = list(
-	  status = function(x) { if (missing(x)) private$jaspObject$status else private$jaspObject$status <- x }
+	  status = function(x) { if (missing(x)) private$jaspObject$status else private$jaspObject$status <- x },
+	  info   = function(x) { if (missing(x)) private$jaspObject$info   else private$jaspObject$info   <- x }
```

Comment:
Would you ever want to request `info`? For `status` I can see that being useful.

Reply:
Well, maybe not, but I like to reason from possibilities and not from constraints ;)
Because there is nothing bad about allowing introspection into this info right?
Maybe if you are working in R and are like "what should this container I got from this other module be anyway?". 

But mostly, I see nothing better resulting from not allowing it to be requested.

Reply:
I'm mainly concerned about having a consistent API for developers.
As an example:
`$addCitation("Some citation")`
`$info <- "Some info"`

It'd be good to be consistent in what we think ought to be active bindings and what ought to be functions.

Reply:
But you can add multiple citations right? And only one "info".

Reply:
`$citations <- c("cite1", "cite2")`
But it was just an example, `setError("error")` does the same...

What is the rule we can apply to differentiate being an active binding and a function?

Reply:
I would go more for active bindings I think. Because it doesn't really matter what happens on the background to R anyway and I like the syntax of being able to set and get so easily. 
But you can veto that real hard if you like.

Reply:
> What is the rule we can apply to differentiate between an active binding and a function?

We don't have any, but we probably should.

Right now everything that can be set but not read is done with `$addXXX` right? E.g., `$setError`, `$addCitation`. I'm not really seeing any use case for reading the info text if it's should only be set. So without a use case for `x <- obj$info` I'd make this `obj$addInfo("...")`, and I'd also add a check that its argument is a character of length 1 (although maybe you now do that in C++?).

On the other hand, if I'd use `jaspResults` objects in R I'd prefer `obj$info` to print the info and `obj$citations` to print the citations. So in that use case, I'd prefer active methods. Active methods might make it easier to write bad R code though.

Perhaps a middle ground is to only allow `x <- obj$info` in interactive sessions (for module development) using something like `if (jaspResultsCalledFromJASP) stop("that's illegal") else return(x$...$info)`.

Reply:
>On the other hand, if I'd use jaspResults objects in R I'd prefer obj$info to print the info and obj$citations to print the citations. So in that use case, I'd prefer active methods. Active methods might make it easier to write bad R code though.
But why would having active bindings lead to bad code? 
And I think having more active bindings (or properties) is more in line also with how I would work in QML.

> Perhaps a middle ground is to only allow `x <- obj$info` in interactive sessions (for module development)
I guess we could do this but I would vote against it based:
1. It makes the code more complicated just to break something in a specific scenario (and the unittests would still accept these...) 
2. People are already confused when they realize jaspResults isn't even a package in JASP. If it also behaves differently, that would make it quite annoying.

My arguments *for* active bindings is: easier debugging because you can check in the code what was put in there. And I would extend this to `setError` and `addCitations` as well.

But maybe you guys can indicate to me what kind of bad code you are afraid these bindings will lead to? Because if that is scary enough ;)

Reply:
I think we were worried about something like:
```r
if (table$citations != "") # means the table code didn't crash but now imagine the comment isn't there.
```
I remember that @TimKDJ once encountered something similar, where the analysis checked if a particular element of the state existed which meant that some other function had crashed or not.

But actually, I agree with your arguments. For citations, we should just make it into an active binding. For errors, this is a bit complicated. `setError` takes a string, and `hasError` returns a bool. If you translate this to an active binding it looks a bit odd. (We could create a custom class for that which on the R side is just a bool with an attribute that contains an error message but it's still odd that the setter takes a string and the getter returns a bool). Perhaps we should make a concrete list of what methods should be active functions. The whole `getColumnName`, `setColumnName`, `setColumnTitle`, `getColumnTitle`, ... extravaganza could just be done with active bindings (and hopefully `colnames(...)` and `colnames(...)<-` wrappers.

Either way, if we decide to completely switch to one of the two, I agree that active bindings makes the most sense. It does require adjusting all analyses though.

Reply:
Yeah I actually implemented support for all thos `colNames` things long long ago in the first release of jaspTable... They just aren't present in th R6 wrapper. And it kind of seemed that the whole `addColumnInfo` was more workable anyway.

The error thing could be an active binding that sets and gets the errormsg. But sets `hasError` to true as well under water. That would make it nicer no?
Ill make an issue for this point and link to this discussion.

Reply:
The example looks like a pretty bad check... But Im sure we would catch most of these with code review. And you know giving people an easy way to check stuff didn't crash or fail. Which the whole `setError` thing is for right?